WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,166 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using System.Threading;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		public bool RequestEnd { get; set; }
```

Comment:
The variable can be used by multiple threads (UI thread and whatever thread that `Task.Run` gives you) and yet it is not synchronized. Unless I'm missing something, I think it should be either `volatile` or `Interlocked` class should be used to get value of the variable.

Reply:
I got asked to add `.ObserveOn(RxApp.MainThreadScheduler)` for UI syncronization.
Could that be a solution for this problem?

Reply:
No. That does not change anything on the thing I said.

Reply:
Not that you have the `ScanningTaskLock`, it can be done with a `cancellationToken` as well.